src/data/resolvers/colony.ts

```diff
@@ -404,20 +406,46 @@ export const colonyResolvers = ({
         return null;
       }
     },
-    async canMintNativeToken({ colonyAddress }) {
+    async canColonyMintNativeToken({ colonyAddress }) {
       const colonyClient = await colonyManager.getClient(
         ClientType.ColonyClient,
         colonyAddress,
       );
-      // fetch whether the user is allowed to mint tokens via the colony
+      const { tokenClient } = colonyClient;
+      const tokenAuthorityAddress = tokenClient.authority();
+      const tokenAuthorityContract = new Contract(
+        tokenAuthorityAddress,
+        tokenAuthorityABI,
+        colonyManager.signer,
+      );
+      const mintSighash = tokenClient.interface.functions.mint.sighash;
+
+      // fetch whether the colony is allowed to mint tokens
       let canMintNativeToken = true;
       try {
-        await colonyClient.estimate.mintTokens(bigNumberify(1));
+        canMintNativeToken = await tokenAuthorityContract.canCall(
+          colonyAddress,
+          tokenClient.address,
+          mintSighash,
+        );
       } catch (error) {
         canMintNativeToken = false;
       }
       return canMintNativeToken;
     },
+    async canUserMintNativeToken({ colonyAddress }) {
+      const colonyClient = await colonyManager.getClient(
+        ClientType.ColonyClient,
+        colonyAddress,
+      );
+      // fetch whether the user is allowed to mint tokens via the colony
+      try {
+        await colonyClient.estimate.mintTokens(bigNumberify(1));
```

Comment:
As talked on discord, how come the `canUserMintNativeToken` doesn't include the user address? Could you please clarify this - for reference.

Reply:
Talked with Raul about this. `.mintTokens` doesn't include the user address because the contracts already extract the address from the wallet that is sending the transaction to check if they can mint or not. 

It's a safe way of checking that the user has permission to mint, otherwise, if we added an address param to the method call any user could just plug in an address with permission that is not necessarily their own and mint tokens like nobody business.